drivers/input/input_renesas_ra_ctsu.c

```diff
@@ -0,0 +1,704 @@
+	struct renesas_ra_ctsu_group_data *data = dev->data;
+
+	if (p_args->event == CTSU_EVENT_SCAN_COMPLETE) {
+		k_work_submit(&data->reading_work);
+	}
+
+	k_sem_give(&ctsu_data->scanning);
+}
+
+#define POLLING_INTERVAL_MS K_MSEC(CONFIG_INPUT_RENESAS_RA_CTSU_POLLING_INTERVAL_MS)
+#define STABILIZATION_MS    K_USEC(CONFIG_INPUT_RENESAS_RA_CTSU_STABILIZATION_TIME_US)
+
+static void renesas_ra_ctsu_drv_handler(void *arg0, void *arg1, void *arg2)
+{
+	ARG_UNUSED(arg1);
+	ARG_UNUSED(arg2);
+	const struct device *ctsu_dev = (const struct device *)arg0;
+	struct renesas_ra_ctsu_data *ctsu_data = ctsu_dev->data;
+
+	while (true) {
+		struct ctsu_scan_msg *msg = k_queue_get(&ctsu_data->scan_q, K_FOREVER);
+		struct st_touch_instance *p_instance = msg->p_instance;
+		fsp_err_t err;
+
+		k_sem_reset(&ctsu_data->scanning);
+		err = p_instance->p_api->scanStart(p_instance->p_ctrl);
+		if (err == FSP_SUCCESS) {
+			k_sem_take(&ctsu_data->scanning, K_FOREVER);
+		}
+
+		sys_mem_blocks_free(&scan_msg_allocator, 1, (void **)&msg);
+
+		k_sleep(STABILIZATION_MS);
+	}
+}
+
+static void renesas_ra_ctsu_group_sampling_handler(struct k_timer *timer)
+{
+	struct renesas_ra_ctsu_group_data *data =
+		CONTAINER_OF(timer, struct renesas_ra_ctsu_group_data, sampling_timer);
+	const struct device *ctsu_dev = k_timer_user_data_get(timer);
+	struct renesas_ra_ctsu_data *ctsu_data = ctsu_dev->data;
+	struct ctsu_scan_msg *msg;
+
+	if (sys_mem_blocks_alloc(&scan_msg_allocator, 1, (void **)&msg) != 0) {
+		return;
+	}
+
+	msg->p_instance = (void *)&data->touch_instance;
+	k_queue_append(&ctsu_data->scan_q, msg);
+}
+
+#if DT_HAS_COMPAT_STATUS_OKAY(renesas_ra_ctsu_button)
+static void renesas_ra_ctsu_group_buttons_read(const struct device *dev)
+{
+	const struct renesas_ra_ctsu_group_cfg *cfg = dev->config;
+	struct renesas_ra_ctsu_group_data *data = dev->data;
+
+	if (cfg->num_button == 0) {
+		return;
+	}
+
+	if (*data->p_button_status != 0) {
+		uint64_t tmp_status = *data->p_button_status;
+
+		while (tmp_status != 0) {
+			int num = u64_count_trailing_zeros(tmp_status);
+			struct renesas_ra_ctsu_device_cb *p_button_cb = &data->p_button_cb[num];
+
+			p_button_cb->device_cb(p_button_cb->dev, NULL);
+			WRITE_BIT(tmp_status, num, 0);
+		}
+	}
+}
+#endif /* DT_HAS_COMPAT_STATUS_OKAY(renesas_ra_ctsu_button) */
+
+#if DT_HAS_COMPAT_STATUS_OKAY(renesas_ra_ctsu_slider)
+static void renesas_ra_ctsu_group_sliders_read(const struct device *dev)
+{
+	const struct renesas_ra_ctsu_group_cfg *cfg = dev->config;
+	struct renesas_ra_ctsu_group_data *data = dev->data;
+
+	if (cfg->num_slider == 0) {
+		return;
+	}
+
+	for (int i = 0; i < cfg->num_slider; i++) {
+		uint16_t slider_position = data->p_slider_position[i];
+
+		if (slider_position != UINT16_MAX) {
+			struct renesas_ra_ctsu_device_cb *p_slider_cb = &data->p_slider_cb[i];
+
+			p_slider_cb->device_cb(p_slider_cb->dev, &slider_position);
+		}
+	}
+}
+#endif /* DT_HAS_COMPAT_STATUS_OKAY(renesas_ra_ctsu_slider) */
```

Comment:
move the `#if` condition inside the function body and let it be a no-op when this is not enabled, then you can remove the conditionals from `renesas_ra_ctsu_group_reading_handler` and always call these, they'll just get optimized out when they are not needed